src/YobotBrokerV2.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.11;
+
+struct Order { // wei per token is: amountDeposited / quantity
+    uint128 quantity; // num NFTs the user wants
+    uint128 amountDeposited; // wei the user has deposited
+}
+
+struct Drop {
+    address token; // NFT contract
+    uint128 lockStartTime; // timestamp when lock starts
+    uint128 lockLength; // in seconds
+}
+
+import {IERC721} from "./interfaces/IERC721.sol";
+
+
+/// @title YobotBrokerV2
+/// @author Andreas Bigger, Artbotter
+/// @notice Permissionless Minting Broker
+contract YobotBrokerV2 is AccessControlled {
+    uint128 public constant MAX_LOCK_LENGTH = 4 hours;
+
+    Drop[] public drops;
+    mapping (uint256 => mapping (address => Order)) dropIdToUserOrder; // dropId -> user -> order
+    uint256 yoBotBalance;
+
+    event DropInfo(uint256 indexed dropIndex, address token, uint128 _lockStartTime, uint128 _lockLength);
+    event OrderInfo(uint256 indexed dropIndex, address indexed user, uint128 quantity, uint128 amountDeposited);
+    event OrderFilled(uint256 indexed dropIndex, address indexed user, uint256 tokenId);
+
+    // OPERATOR FUNCTIONS
+
+    function createNewDrop(address _token, uint128 _lockStartTime, uint128 _lockLength) external onlyOperator returns (uint256) {
+        require(_lockStartTime > block.timestamp, 'bad lock start time');
```

Comment:
clarification q: why do we use lockStartTime & lockLength as opposed to a timestamped drop deadline?